pandas/tests/io/parser/compression.py

```diff
@@ -64,83 +68,36 @@ def test_zip(self):
                 pytest.raises(zipfile.BadZipfile, self.read_csv,
                               f, compression='zip')
 
-    def test_gzip(self):
-        import gzip
-
-        with open(self.csv1, 'rb') as data_file:
-            data = data_file.read()
-            expected = self.read_csv(self.csv1)
-
-        with tm.ensure_clean() as path:
-            tmp = gzip.GzipFile(path, mode='wb')
-            tmp.write(data)
-            tmp.close()
-
-            result = self.read_csv(path, compression='gzip')
-            tm.assert_frame_equal(result, expected)
-
-            with open(path, 'rb') as f:
-                result = self.read_csv(f, compression='gzip')
-                tm.assert_frame_equal(result, expected)
-
-        with tm.ensure_clean('test.gz') as path:
-            tmp = gzip.GzipFile(path, mode='wb')
-            tmp.write(data)
-            tmp.close()
-            result = self.read_csv(path, compression='infer')
-            tm.assert_frame_equal(result, expected)
-
-    def test_bz2(self):
-        import bz2
+    @pytest.mark.parametrize('compress_type, compress_method, ext', [
+        ('gzip', gzip.GzipFile, 'gz'),
+        ('bz2', bz2.BZ2File, 'bz2'),
+        pytest.param('xz', lzma.LZMAFile, 'xz',
+                     marks=td.skip_if_no_lzma)
+    ])
+    def test_other_compression(self, compress_type, compress_method, ext):
 
         with open(self.csv1, 'rb') as data_file:
             data = data_file.read()
             expected = self.read_csv(self.csv1)
 
         with tm.ensure_clean() as path:
-            tmp = bz2.BZ2File(path, mode='wb')
+            tmp = compress_method(path, mode='wb')
             tmp.write(data)
             tmp.close()
 
-            result = self.read_csv(path, compression='bz2')
+            result = self.read_csv(path, compression=compress_type)
             tm.assert_frame_equal(result, expected)
 
-            pytest.raises(ValueError, self.read_csv,
-                          path, compression='bz3')
+            if compress_type == 'bz2':
+                pytest.raises(ValueError, self.read_csv,
+                              path, compression='bz3')
```

Comment:
typo

Reply:
But the reason it's 'bz3' and not 'bz2' is that it's supposed to raise a ValueError

Reply:
ahh ok

Reply:
hmm i would put that testing of the error co diction in a separate test

Reply:
actually on 2nd thought, this is ok.